lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxBuilderImpl.scala

```diff
@@ -41,14 +41,18 @@ class OutBoxBuilderImpl(_txB: UnsignedTransactionBuilderImpl) extends OutBoxBuil
 
     if(token.getMintingBoxR7 != null){
       _registers ++= Array(token.getMintingBoxR7)
-    }
-    if(token.getMintingBoxR8 != null){
-      _registers ++= Array(token.getMintingBoxR8)
-    }
-    if(token.getMintingBoxR9 != null){
-      _registers ++= Array(token.getMintingBoxR9)
-    }
 
+      if(token.getMintingBoxR8 != null){
+        _registers ++= Array(token.getMintingBoxR8)
+
+
+        if(token.getMintingBoxR9 != null){
+          _registers ++= Array(token.getMintingBoxR9)
+        }
+
+      }
+
+    }
```

Comment:
@mgpai22, please think about full validation. Whati token.getMintingBoxR7 == null, but token.getMintingBoxR8 or token.getMintingBoxR9 are not. This shouldn't be silently swallowed, it is better to throw a IllegalArgumentException.

Reply:
Good point, the old code caused bugs for me before while testing the modified eip4 standard since there weren't exceptions thrown.